scripts/check_changeset_completeness.ts

```diff
@@ -0,0 +1,49 @@
+import getReleasePlan from '@changesets/get-release-plan';
+import { GitClient } from './components/git_client.js';
+import { readPackageJson } from './components/package-json/package_json.js';
+import { EOL } from 'os';
+
+const gitClient = new GitClient();
+
+const baseRef = process.argv[2];
+if (baseRef === undefined) {
+  throw new Error('No base ref specified for changeset completeness check');
+}
+
+const releasePlan = await getReleasePlan(process.cwd());
+
+const packagesWithChangeset = new Set(
+  releasePlan.releases.map((release) => release.name)
+);
+
+const changedFiles = await gitClient.getChangedFiles(baseRef);
+
+const modifiedPackageDirs = new Set<string>();
+
+changedFiles
+  .filter((changedFile) => changedFile.startsWith('packages/'))
+  .forEach((changedPackageFile) => {
+    modifiedPackageDirs.add(
+      changedPackageFile.split('/').slice(0, 2).join('/')
+    );
+  });
+
+const packagesMissingChangesets = [];
+for (const modifiedPackageDir of modifiedPackageDirs) {
+  const { name: modifiedPackageName, private: isPrivate } =
+    await readPackageJson(modifiedPackageDir);
+  if (isPrivate) {
+    continue;
+  }
+  if (!packagesWithChangeset.has(modifiedPackageName)) {
+    packagesMissingChangesets.push(modifiedPackageName);
+  }
+}
+
+if (packagesMissingChangesets.length > 0) {
+  throw new Error(
+    `The following packages have changes but are not included in any changeset:${EOL}${EOL}${packagesMissingChangesets.join(
+      EOL
+    )}${EOL}${EOL}Add a changeset using 'npx changeset add' or 'npx changeset add --empty' if this is a non-functional change.`
```

Comment:
I think message now should clarify that `--empty` is allowed if only private packages are changed.

Reply:
Things like adding tests or linting could also have an empty changeset. Although maybe those cases are small enough that it's not worth mentioning

Reply:
Good point.
We could perhaps have some predicate function in the algorithm that narrows down this classification.

something like

```
const shouldHaveNonEmptyChangeSet: () : 'yes'|'no'|'maybe' => {

if (API.md changed) {
 return yes
}
if (Only Test Changed) {
return no
}

return maybe;
}
```

Reply:
I'll leave this as a future optimization if needed. For now, I just removed the empty changeset suggestion. Having an unnecessary version bump is much better than a missing version bump.